# HG changeset patch
# User Roman Neuhauser <neuhauser@sigpipe.cz>
# Date 1299598633 -3600
# Node ID 8200ac47ff3963886a60e51016fdc7b95154bb06
# Parent  e7007dfcf852748c12a77e4901a99a91b46f6359
test archive_read_close() (doubled, even) on an unopened archive

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -249,6 +249,7 @@
 	libarchive/test/test_acl_posix1e.c			\
 	libarchive/test/test_archive_api_feature.c		\
 	libarchive/test/test_archive_clear_error.c		\
+	libarchive/test/test_archive_read_close_twice.c		\
 	libarchive/test/test_archive_set_error.c		\
 	libarchive/test/test_archive_read_set_filter_option.c	\
 	libarchive/test/test_archive_read_set_format_option.c	\
diff --git a/libarchive/test/CMakeLists.txt b/libarchive/test/CMakeLists.txt
--- a/libarchive/test/CMakeLists.txt
+++ b/libarchive/test/CMakeLists.txt
@@ -14,6 +14,7 @@
     test_acl_posix1e.c
     test_archive_api_feature.c
     test_archive_clear_error.c
+    test_archive_read_close_twice.c
     test_archive_set_error.c
     test_archive_read_set_filter_option.c
     test_archive_read_set_format_option.c
diff --git a/libarchive/test/test_archive_read_close_twice.c b/libarchive/test/test_archive_read_close_twice.c
new file mode 100644
--- /dev/null
+++ b/libarchive/test/test_archive_read_close_twice.c
@@ -0,0 +1,43 @@
+/*-
+ * Copyright (c) 2003-2010 Tim Kientzle
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR(S) ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR(S) BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "test.h"
+__FBSDID("$FreeBSD$");
+
+
+DEFINE_TEST(test_archive_read_close_twice)
+{
+	struct archive* a = archive_read_new();
+
+	assertEqualInt(0, archive_read_close(a));
+	assertEqualInt(0, archive_errno(a));
+	assertEqualString(NULL, archive_error_string(a));
+
+	assertEqualInt(0, archive_read_close(a));
+	assertEqualInt(0, archive_errno(a));
+	assertEqualString(NULL, archive_error_string(a));
+
+	archive_read_finish(a);
+}
